datafusion/core/tests/sql/recursive_cte.rs

```diff
@@ -0,0 +1,130 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use super::*;
+use datafusion::prelude::{CsvReadOptions, SessionContext};
+use datafusion_common::test_util::datafusion_test_data;
+
+#[tokio::test]
+async fn recursive_cte_alias_instability() -> Result<()> {
```

Comment:
Is there any reason this is not a slt test?

I also don't understand the `instability` in the name of the function. What is unstable?